SpatialGDK/Source/SpatialGDKServices/Private/SpatialPackageService.cpp

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialPackageService.h"
+
+#include "Async/Async.h"
+#include "HAL/FileManager.h"
+#include "SpatialGDKServicesConstants.h"
+#include "SpatialGDKServicesModule.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialPackageService);
+
+namespace
+{
+	bool DownloadPackage(const FSpatialPackageDescriptor& Package)
+	{
+		UE_LOG(LogSpatialPackageService, Log, TEXT("Downloading package: '%s'"), *Package.ToString());
+
+		const FString Args = FString::Printf(TEXT("package retrieve %s %s %s %s --unzip"),
+			*Package.Type, *Package.Name, *Package.Version, *Package.GetExecutableDir());
+
+		FString Result;
+		int32 ExitCode;
+		FSpatialGDKServicesModule::ExecuteAndReadOutput(*SpatialGDKServicesConstants::SpatialExe, Args,
+			*SpatialGDKServicesConstants::GDKProgramPath, Result, ExitCode);
+
+		if (ExitCode != 0)
+		{
+			UE_LOG(LogSpatialPackageService, Error, TEXT("Could not download package '%s' from package service"), *Package.ToString());
+			return false;
+		}
+
+		UE_LOG(LogSpatialPackageService, Log, TEXT("Downloaded package: %s"), *Result);
+		return true;
+	}
+}
+
+FString FSpatialPackageDescriptor::ToString() const
+{
+	return FString::Printf(TEXT("%s %s %s"), *Type, *Name, *Version);
+}
+
+FString FSpatialPackageDescriptor::GetExecutable() const
```

Comment:
Nit : GetExecutablePath

Reply:
I'm worried about the assumption that this will always be an executable, considering that using this for the runtime in such a way makes this the only exe you can get.

Reply:
Yeah, I thought it's an acceptable assumption for 2020.2 which would have just required to download two packages that have to contain an executable (Runtime + Inspector). We can certainly make this more generic, if we expect to use the package service for anything else that's not an executable in the future.